File: langs/python/src/vmx/lifecycle/status.py
"""ConstructionStatus — the five states of a VMx viewmodel's lifecycle.

See spec/02-lifecycle.md for the full state-machine contract.
"""

from __future__ import annotations

from enum import IntEnum


class ConstructionStatus(IntEnum):
    """Five-state lifecycle enum.

    Values match the C# counterpart exactly so that JSON fixtures and
    wire serialisation round-trip cleanly across language boundaries.
    """

    #: Terminal state.  Once entered, cannot leave.
    DISPOSED = 0

    #: Transient state during destruct().
    DESTRUCTING = 1

    #: Initial state of a freshly built VM.
    DESTRUCTED = 2

    #: Transient state during construct().
    CONSTRUCTING = 3

    #: Ready-to-use state.
    CONSTRUCTED = 4

File: langs/typescript/src/lifecycle/status.ts
/**
 * ConstructionStatus — five-state lifecycle enum.
 *
 * Values match the Python and C# counterparts so that JSON fixtures
 * round-trip cleanly across language boundaries.
 *
 * See spec/02-lifecycle.md for the full state-machine contract.
 */
export enum ConstructionStatus {
  /** Terminal state. Once entered, cannot leave. */
  Disposed = 0,
  /** Transient state during destruct(). */
  Destructing = 1,
  /** Initial state of a freshly built VM. */
  Destructed = 2,
  /** Transient state during construct(). */
  Constructing = 3,
  /** Ready-to-use state. */
  Constructed = 4,
}
